KeenClient/KeenClient.m

```diff
@@ -1076,144 +665,18 @@ - (void)runAsyncMultiAnalysisWithQueries:(NSArray *)keenQueries block:(void (^)(
     });
 }
 
-# pragma mark Sync methods
-
 - (void)runQuery:(KIOQuery *)keenQuery completionHandler:(void (^)(NSData *data, NSURLResponse *response, NSError *error))completionHandler {
-    BOOL hasQueryWithMaxAttempts = [self hasQueryReachedMaxAttempts:keenQuery];
-
-    if(hasQueryWithMaxAttempts) {
-        KCLog(@"Not running query because it failed over %d times", self.maxQueryAttempts);
-    } else {
-        NSString *urlString = [NSString stringWithFormat:@"%@/%@/projects/%@/queries/%@",
-                               kKeenServerAddress, kKeenApiVersion, self.projectID, keenQuery.queryType];
-        KCLog(@"Sending request to: %@", urlString);
-        NSURL *url = [NSURL URLWithString:urlString];
-        NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url cachePolicy:NSURLRequestUseProtocolCachePolicy timeoutInterval:30.0f];
-        [request setHTTPMethod:@"POST"];
-        [request setValue:@"application/json" forHTTPHeaderField:@"Accept"];
-        [request setValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
-        [request setValue:self.readKey forHTTPHeaderField:@"Authorization"];
-        [request setValue:[NSString stringWithFormat:@"%lud",(unsigned long) [[keenQuery convertQueryToData] length]] forHTTPHeaderField:@"Content-Length"];
-        [request setValue:kKeenSdkVersionWithPlatform forHTTPHeaderField:kKeenSdkVersionHeader];
-        [request setHTTPBody:[keenQuery convertQueryToData]];
-
-        NSURLSession *session = [self sharedUrlSession];
-        [[session dataTaskWithRequest:request completionHandler:completionHandler] resume];
-    }
+    [self.network runQuery:keenQuery
```

Comment:
Same question here. I suppose the hasQueryWithMaxAttempts stuff is handled in network module now, right?